### ISSUE TITLE
Why need support.valid_dl and query.valid_dl

### ISSUE THREAD

**Issue by WNZhao1988** [amazon-science/few-shot-baseline#2]
Hi, 
I'm quite interested in your proposed transductive few-shot learning. After reading your code, I have some questions.

1. Why we need to use support.valid_dl for support-initialization instead of all support samples?
2. Why we cannot use all query samples instead of query.train_dl to calculate entropy regularizer? I think one reason may be as you mentioned in your paper that if we use all query samples for entropy regularizer, this may cause an imbalance between the support set and entropy regularizer calculated by query set. But I wonder if another reason is that this would cause overfitting in the query set (or what we can call test set)?

Actually, although I think transductive information is useful and sounds reasonable, I still suspect we use query during finetuning is a kind of data leakage. Could you provide more information or explain more about this to alleviate my concern?

Thank you!
